core/src/snapshot/postprocessers/postprocess.rs

```diff
@@ -0,0 +1,198 @@
+        }
+
+        // handle the inside of the expression
+        let inside = match expression.get(2..expression.len() - 2) {
+            Some(x) => x.to_string(),
+            None => "".to_string(),
+        };
+
+        let inner_tokens = tokenize(&inside);
+        return !inner_tokens.iter().any(|tk| classify_token(tk) == TokenType::Operator);
+    }
+
+    let mut cleaned: String = line.to_owned();
+    println!("cleaned");
+
+    // get the nth index of the first open paren
+    let nth_paren_index = match cleaned.match_indices('(').nth(paren_index) {
+        Some(x) => x.0,
+        None => return cleaned,
+    };
+    
+    //find it's matching close paren
+    let (paren_start, paren_end, found) =
+        find_balanced_encapsulator(&cleaned[nth_paren_index..], ('(', ')'));
+
+    // remove paren_start if can't find a closing one
+    let mut found_match = found;
+
+    if !found_match {
+        cleaned.remove(nth_paren_index);
+        found_match = true;
+    }
+    // add the nth open paren to the start of the paren_start
+    let paren_start = paren_start + nth_paren_index;
+    let paren_end = paren_end + nth_paren_index;
+
+    if let true = found_match {
+        // get the logical expression including the char before the parentheses (to detect casts)
+        let logical_expression = match paren_start {
+            0 => match cleaned.get(paren_start..paren_end + 1) {
+                Some(expression) => expression.to_string(),
+                None => cleaned[paren_start..paren_end].to_string(),
+            },
+            _ => match cleaned.get(paren_start - 1..paren_end + 1) {
+                Some(expression) => expression.to_string(),
+                None => cleaned[paren_start - 1..paren_end].to_string(),
+            },
+        };
+
+        println!("logical_expression: {}",  logical_expression); 
+
+        println!("checking if parenthesis necessary");
+        // check if the parentheses are unnecessary and remove them if so
+        if are_parentheses_unnecessary(&logical_expression) {
+            println!("parenthesis are unnecessary");
+            cleaned.replace_range(
+                paren_start..paren_end,
+                match logical_expression.get(2..logical_expression.len() - 2) {
+                    Some(x) => x,
+                    None => "",
+                },
+            );
+            println!("cleaned after replace_range {}", cleaned);
+
+
+            // remove double negation, if one was created
+            if cleaned.contains("!!") {
+                println!("cleared double negation");
+                cleaned = cleaned.replace("!!", "");
+            }
+
+            // recurse into the next set of parentheses
+            // don't increment the paren_index because we just removed a set
+            cleaned = simplify_parentheses(&cleaned, paren_index);
+        } else {
+            // remove double negation, if one exists
+            if cleaned.contains("!!") {
```

Comment:
this if statement might be redundant, since replace will replace nothing if `!!` doesnt exist in the string.